rs/sns/swap/src/swap.rs

```diff
@@ -1618,155 +1621,42 @@ impl Swap {
         // be accessed in O(1) time.
         let mut claimable_neurons_index = btreemap! {};
 
-        // The NeuronParameters that will be used to create neurons.
-        let mut neuron_parameters = vec![];
+        // The `NeuronRecipe`s that will be used to create neurons.
+        // We are converting `SnsNeuronRecipe`s to a type with a type with a similar name,
+        // `NeuronRecipe`, as this is the type expected by the SNS Governance canister.
+        let mut neuron_recipes = vec![];
 
         for recipe in &mut self.neuron_recipes {
-            // TODO(NNS1-3207): This match will need to be changed to evaluate to a Participant
-            let (hotkey, controller, source_nns_neuron_id) = match recipe.investor.as_ref() {
-                Some(Investor::Direct(DirectInvestment { buyer_principal })) => {
-                    let parsed_buyer_principal = match string_to_principal(buyer_principal) {
-                        Some(p) => p,
-                        // principal_str should always be parseable as a PrincipalId as that is enforced
-                        // in `refresh_buyer_tokens`. In the case of a bug due to programmer error, increment
-                        // the invalid field. This will require a manual intervention via an upgrade to correct
-                        None => {
-                            sweep_result.invalid += 1;
-                            continue;
-                        }
-                    };
-
-                    (None, parsed_buyer_principal, None)
-                }
-                Some(Investor::CommunityFund(cf_investment)) => {
```

Comment:
NCR;

Could you remind me why the code is removed from this function? Are we now making the distinction between NF / direct participants elsewhere? Is it in `recipe.to_neuron_recipe`?